MaloWBotLK/Functions.lua

```diff
@@ -292,6 +296,59 @@ function mb_getMySpecName()
 	return name
 end
 
+-- Scans through all bag slots looking for item by name
+function mb_GetItemLocation(itemName)
+	for bag = 0, 4 do
+		for slot = 1, GetContainerNumSlots(bag) do
+			local texture = GetContainerItemInfo(bag, slot)
+			if texture ~= nil then
+				local name = GetItemInfo(mb_GetItemStringFromItemLink(GetContainerItemLink(bag, slot)))
+				if itemName == name then
+					return bag, slot
+				end
+			end
+		end
+	end
+	return nil
+end
 
+-- Converts an ItemLink to an ItemString
+function mb_GetItemStringFromItemLink(itemLink)
+	local found, _, itemString = string.find(itemLink, "^|%x+|H(.+)|h%[.+%]")
+	return itemString
+end
 
+-- Uses item in bag by name
+function mb_UseItem(itemName)
+	local bag, slot = mb_GetItemLocation(itemName)
+	if bag ~= nil then
+		UseContainerItem(bag, slot)
+		return true
+	end
+	return false
+end
+-- Uses mage food if missing mana and not in combat
+function mb_DrinkIfGood()
+	if mb_ShouldStopDrinking() then
+		return false
+	end
 
+	if mb_unitPowerPercentage("player") < 50 and not UnitAffectingCombat("player") then
+		if UnitBuff("player", "Drink") then
+			return true
+		end
+
+		mb_UseItem("Conjured Mana Strudel")
```

Comment:
In order to work at every level I would like this to not be hard-coded but instead pulled from the Config.lua.
I would suggest adding 
mb_waterConfig = {}
table.insert(mb_waterConfig, "Conjured Mana Strudel")
in Config.lua, and then changing this here to 
for _, waterName in pairs(mb_waterConfig) do
    if mb_UseItem(waterName) then
        return true
    end
end

That way players of different levels or setups that don't have a mage can easily add other water-items to the config to get them to drink those.